src/azure-cli/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -301,19 +301,20 @@ def validate_subnet(cmd, namespace):
 
 
 def validate_vault_id(entity_type):
-
     def _validate(ns):
         from azure.keyvault.key_vault_id import KeyVaultIdentifier
-        name = getattr(ns, entity_type.replace('deleted', '') + '_name', None)
+
+        pure_entity_type = entity_type.replace('deleted', '')
```

Comment:
is this a service behavior change?

Reply:
@yungezz No, this is a bug, other normal service behavior will not be affected.

Bug detail:
In command `az keyvault secret show-deleted --id <id>`, id looks like `https://cli-test-kevault-000002.vault.azure.net/deletedsecrets/secret1`, and the previous validator will set a variable named `deletedsecret_name` to store secret name (here the value is `secret1`).
But the command requires param `secret_name`, so the problem described in issue #10846 will occur.

Reply:
got it, thanks.